internal/provider/webhook/resource_webhook.go

```diff
@@ -0,0 +1,530 @@
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/hashicorp/terraform-provider-hcp/internal/clients"
+	"github.com/hashicorp/terraform-provider-hcp/internal/hcpvalidator"
+)
+
+// Ensure provider defined types fully satisfy framework interfaces.
+var _ resource.Resource = &resourceWebhook{}
+var _ resource.ResourceWithImportState = &resourceWebhook{}
+
+func NewWebhookResource() resource.Resource {
+	return &resourceWebhook{}
+}
+
+type resourceWebhook struct {
+	client *clients.Client
+}
+
+func (r *resourceWebhook) Metadata(_ context.Context, req resource.MetadataRequest, resp *resource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_webhook"
+}
+
+func (r *resourceWebhook) Schema(_ context.Context, _ resource.SchemaRequest, resp *resource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		Description: "The webhook resource manages a HCP webhook, used to notify external systems about a " +
+			"project resource's lifecycle events",
+
+		Attributes: map[string]schema.Attribute{
+			"name": schema.StringAttribute{
+				Required:    true,
+				Description: "The webhook's name.",
+				Validators: []validator.String{
+					hcpvalidator.ResourceNamePart(),
+				},
+			},
+
+			"config": schema.SingleNestedAttribute{
+				Required:    true,
+				Description: "The webhook configuration used to deliver event payloads.",
+				Attributes: map[string]schema.Attribute{
+					"url": schema.StringAttribute{
+						Required: true,
+						MarkdownDescription: `The HTTP or HTTPS destination URL that HCP delivers the event payloads to. 
+The destination must be able to use the HCP webhook 
+[payload](https://developer.hashicorp.com/hcp/docs/hcp/admin/projects/webhooks#webhook-payload).`,
+						Validators: []validator.String{
+							hcpvalidator.URL(),
+						},
+					},
+					"hmac_key": schema.StringAttribute{
+						Optional: true,
+						Description: "The arbitrary secret that HCP uses to sign all its webhook requests. This is a" +
+							"write-only field, it is written once and not visible thereafter.",
+						Sensitive: true,
+					},
+				},
+			},
+
+			// Optional fields
+			"project_id": schema.StringAttribute{
+				Description: "The ID of the project to create the webhook under. " +
+					"If unspecified, the webhook will be created in the project the provider is configured with. ",
+				Optional: true,
+				Computed: true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.RequiresReplace(),
+					stringplanmodifier.UseStateForUnknown(),
+				},
+			},
+
+			"description": schema.StringAttribute{
+				Optional:    true,
+				Description: "The webhook's description. Descriptions are useful for helping others understand the purpose of the webhook.",
+			},
+
+			"enabled": schema.BoolAttribute{
+				Optional:            true,
+				Computed:            true,
+				MarkdownDescription: "Indicates if the webhook should receive payloads for the subscribed events.",
```

Comment:
Should receive -> should send

Right?

Reply:
My line of thought for `receive`  is that the product services will emit events that HCP will send the payload via a request made to the webhook, configured with this resource, that will then receive those payloads and act on them. I have been writing the docs (API, developer.hashicorp.com) from this perspective. 

The wording I got from https://registry.terraform.io/providers/integrations/github/latest/docs/resources/repository_webhook#active which is also

Reply:
Sounds good, I guess what matters is we are consistent.